### PR TITLE
Add feature flag span enrichment system tests

### PR DESCRIPTION
**Pull request by sameerank**
## Motivation

Add system tests for feature flag event enrichment (Feature ID 551) that validates APM span chunk tags:
- `ffe_flags_enc`: base64(delta varint encoded serial IDs) for all evaluated flags
- `ffe_subjects_enc`: subject hash → serial IDs mapping (when doLog=true)
- `ffe_runtime_defaults`: flag name → default value mapping (when flag not found in UFC)

**Example span tags:**
```json
{
  "ffe_runtime_defaults": {
    "flag-not-found": "my-default-value"
  },
  "ffe_flags_enc": "ZAgUAg==",
  "ffe_subjects_enc": {
    "4208f8d8017ce252df51f51e8c5b558a": "ZA==",
    "8e526f09f0531278f8bd60f224adf60a": "bAg="
  }
}
```
*In this example: `"ZAgUAg=="` decodes to serial IDs [100, 108, 128, 130]*

JIRA: https://datadoghq.atlassian.net/browse/FFL-2201

## Changes

**New test files:**
- `tests/parametric/test_ffe/test_span_enrichment.py` - Span enrichment tests (17 tests)
- `tests/parametric/test_ffe/span-enrichment-flags.json` - UFC fixture with serialId in splits
- `tests/parametric/test_ffe/utils.py` - Delta varint encoding/decoding utilities + SHA256 hashing

**Manifest updates:**
- Added `missing_feature` entries to language manifests (tests written before tracer implementation)

**Why parametric only (no E2E)?**
- Span enrichment is internal tracer behavior, not weblog behavior
- Parametric tests provide precise control for edge cases (limits, child span propagation)

## ⚠️ Important: Span Context for Span Enrichment Tests

**The `ffe/evaluate` endpoint now accepts an optional `span_id` parameter.**

This is required for span enrichment tests because:
1. The parametric server's `/trace/span/start` creates spans but does NOT activate them in the tracer scope
2. When `/ffe/evaluate` runs as a separate HTTP request, `tracer.scope().active()` returns null
3. The `span_id` parameter tells the server to activate the specified span during flag evaluation

**Usage pattern:**
```python
with test_library.dd_start_span(name="test-span", service="test-service") as span:
    test_library.ffe_evaluate(
        flag="my-flag",
        variation_type="BOOLEAN",
        default_value=False,
        targeting_key="user-123",
        attributes={},
        span_id=span.span_id,  # <-- Required for span enrichment
    )
```

## ✅ Verification Against dd-trace-js Implementation

These tests have been verified against the [dd-trace-js span enrichment implementation](https://github.com/DataDog/dd-trace-js/pull/8343).

**Steps to reproduce:**
```bash
# 1. Clone dd-trace-js and checkout the PR branch
cd /path/to/dd-trace-js
git fetch origin pull/8343/head:ffe-span-enrichment
git checkout ffe-span-enrichment

# 2. Point system-tests to your local dd-trace-js
cd /path/to/system-tests
echo "../dd-trace-js" > binaries/nodejs-load-from-local

# 3. Run the span enrichment tests
TEST_LIBRARY=nodejs ./run.sh PARAMETRIC tests/parametric/test_ffe/test_span_enrichment.py

# 4. Cleanup when done
rm binaries/nodejs-load-from-local
```

**Results (all 17 tests pass):**
```
============================= 17 xpassed in 14.85s =============================
```

<details>
<summary>Full test output</summary>

```
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Delta_Varint::test_encoding_decoding_roundtrip
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Delta_Varint::test_empty_list
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Delta_Varint::test_single_value
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Delta_Varint::test_known_encoding_values
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Delta_Varint::test_large_values
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Default_Fallback::test_flag_not_found_adds_ffe_runtime_defaults_tag
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Default_Fallback::test_ffe_runtime_defaults_value_truncated_at_64_chars
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Default_Fallback::test_max_5_flag_keys_in_ffe_runtime_defaults
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Subjects::test_do_log_false_no_subjects_encoded
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Subjects::test_do_log_true_adds_subjects_encoded
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Subjects::test_subjects_encoded_uses_sha256_hashed_targeting_key
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Subjects::test_single_subject_multiple_flags_combines_serial_ids
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Subjects::test_multiple_subjects_tracked_separately
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Serial_IDs::test_multiple_flags_serial_ids_combined
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Child_Span_Propagation::test_child_span_flag_evaluation_propagates_to_root
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Max_Serial_IDs::test_max_128_serial_ids_enforced
XPASS tests/parametric/test_ffe/test_span_enrichment.py::Test_Span_Enrichment_Max_Subjects::test_max_10_subjects_enforced
```

</details>

## 🧪 Local Testing

```bash
# Point system-tests to your local dd-trace-js
echo "../dd-trace-js" > binaries/nodejs-load-from-local

# Run the tests
TEST_LIBRARY=nodejs ./run.sh PARAMETRIC tests/parametric/test_ffe/test_span_enrichment.py

# Cleanup when done
rm binaries/nodejs-load-from-local
```

## Related PRs

- [dd-trace-js #8343](https://github.com/DataDog/dd-trace-js/pull/8343) - FFE span enrichment implementation
- [openfeature-js-client #269](https://github.com/DataDog/openfeature-js-client/pull/269) - Expose serialId in flagMetadata

## Reviewer checklist

* [ ] Anything but `tests/` or `manifests/` is modified ? I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added, removed or renamed?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/parametric/test_ffe/utils.py
Comment (sameerank) [inline tests/parametric/test_ffe/utils.py:1]: File name enforced by CI rule

https://github.com/DataDog/system-tests/blob/01d133adc8ed57d84b4d75cfe985fb33780ffa07/tests/test_the_test/test_conventions.py#L23

File: tests/parametric/test_ffe/test_span_enrichment.py [hunk 1]
```diff
@@ -0,0 +1,464 @@
+"""Test feature flag span enrichment via parametric tests.
+
+This module tests feature flag event enrichment on APM spans:
+1. Multiple flag evaluations combine serial IDs correctly (feature_flags_encoded)
+2. Child span flag evaluation propagates to root span
+3. Max serial ID limit (128)
+4. Max subjects limit (25)
```
Comment (sameerank) [inline tests/parametric/test_ffe/test_span_enrichment.py]: I just picked these numbers randomly, provisionally so I could write tests. What limits should we actually use?

File: manifests/nodejs.yml [hunk 2]
```diff
@@ -1970,6 +1970,7 @@ manifest:
   tests/parametric/test_dynamic_configuration.py::TestDynamicConfigV2: *ref_4_23_0
   tests/parametric/test_ffe/test_dynamic_evaluation.py::Test_Feature_Flag_Dynamic_Evaluation: *ref_5_75_0
   tests/parametric/test_ffe/test_dynamic_evaluation.py::Test_Feature_Flag_Dynamic_Evaluation::test_ffe_of7_empty_targeting_key: bug (FFL-1730)
+  tests/parametric/test_ffe/test_span_enrichment.py: missing_feature (dd-trace-js#8343)
```
Comment (sameerank) [inline manifests/nodejs.yml]: https://github.com/DataDog/dd-trace-js/pull/8343

**Review by dougqh — APPROVED** [review]

File: tests/parametric/test_ffe/test_span_enrichment.py (hunk 1 above)
Comment (sameerank) [reply tests/parametric/test_ffe/test_span_enrichment.py]: Was discussed in https://github.com/DataDog/dd-trace-js/pull/8343 and I matched up the limits that we assert here with with the ones set in that PR

File: manifests/nodejs.yml [hunk 3]
```diff
@@ -2002,6 +2002,7 @@ manifest:
...
```
[the remark this answers is not in the document]
Comment (sameerank) [reply manifests/nodejs.yml]: Fixed https://github.com/DataDog/system-tests/pull/6828/commits/46d7a05508f7887b93be8d65c75f68a4e310ddac

https://github.com/DataDog/system-tests/blob/46d7a05508f7887b93be8d65c75f68a4e310ddac/manifests/nodejs.yml#L2005
